packages/@cdktf/hcl2cdk/test/astGeneration.test.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { convertToTypescript } from "../lib";
+import schema from "./__fixtures__/schemas/astGeneration";
+
+describe("generate typescript", () => {
```

Comment:
I understand where you are coming from here, but wouldn't it be easier to limit the test to just the expression parsing that you changed or if you want to check the generated code rely on the default testing mechanism we use throughout the hcl2cdk codebase?

Reply:
So, there are two problems with the current system, the biggest one by far being too slow. The reason why this test is in the middle of the two approaches is because we need a bit more context than what I have in `references.test.ts` below. I can still run this in a watch, while ensuring that the generated code is changing the way I expect it to.

The reason this test exists here is two-fold. Firstly because I wanted to have a way to check the generation when the expression being assigned (in the test below, `subnet_ids = ...`) is not within a Terraform templated string, does the code do the right thing? Secondly, because I wanted to have a template for this kind of test. 

I'm ok with having a longer discussion about this kind of test, if you still disagree with it and have a better alternative.

Reply:
![xkcd about competing standards](https://imgs.xkcd.com/comics/standards.png)

I'd like to have a longer conversation about his. I get where you are coming from, I think we can solve this for all the tests (eg by setting an environment variable and not synthesising for watched tests). This would help the entire codebase and not just that part.

Reply:
My only concern about this topic would be to introduce environment variables into the codebase that would change behavior while testing, but I'm sure that's not what you plan to do, @DanielMSchmidt 😄